pkg/applicationmanager/controllers/applicationclone.go

```diff
@@ -543,6 +549,16 @@ func (a *ApplicationCloneController) prepareResources(
 	return tempObjects, nil
 }
 
+func (a *ApplicationCloneController) prepareServiceResource(
+	object runtime.Unstructured,
+) error {
+	err := unstructured.SetNestedField(object.UnstructuredContent(), nil, "spec", "ports")
```

Comment:
Instead of setting ports to nil, should we iterate over all the ports and only unset the targetPort so that k8s will assign a new one?

Reply:
unsetting `nodePort` by default, in case someone wants to keep current behaviour they can apply `stork.libopenstorage.org/skip-modify-resource=true` annotation to sac